appengine-standard-java8/cloudsql/src/main/java/com/example/appengine/cloudsql/ListResults.java

```diff
@@ -0,0 +1,138 @@
+
+  // Blog post retrieval queries
+  final String selectSql =
+      "SELECT posts.post_id, users.user_fullname, posts.timestamp, posts.title, posts.body FROM posts, users "
+          + "WHERE (posts.author_id = users.user_id) AND (posts.body != \"\") ORDER BY posts.post_id DESC";
+
+  // Preformatted HTML
+  String headers =
+      "<!DOCTYPE html><meta charset=\"utf-8\"><h1>Welcome to the App Engine Blog</h1><h3><a href=\"blogpost\">Add a new post</a></h3>";
+  String blogPostDisplayFormat =
+      "<h2> %s </h2> Posted at: %s by %s [<a href=\"/update?id=%s\">update</a>] | [<a href=\"/delete?id=%s\">delete</a>]<br><br> %s <br><br>";
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+    // Retrieve blog posts from Cloud SQL database and display them
+
+    PrintWriter out = resp.getWriter();
+
+    out.println(headers); // Print HTML headers
+
+    try (ResultSet rs = conn.prepareStatement(selectSql).executeQuery()) {
+      Map<Integer, Map<String, String>> storedPosts = new HashMap<>();
+
+      while (rs.next()) {
+        Map<String, String> blogPostContents = new HashMap<>();
+
+        // Store the particulars for a blog in a map
+        blogPostContents.put("author", rs.getString("users.user_fullname"));
+        blogPostContents.put("title", rs.getString("posts.title"));
+        blogPostContents.put("body", rs.getString("posts.body"));
+        blogPostContents.put("publishTime", rs.getString("posts.timestamp"));
+
+        // Store the post in a map with key of the postId
+        storedPosts.put(rs.getInt("posts.post_id"), blogPostContents);
+      }
+
+      // Iterate the map and display each record's contents on screen
+      storedPosts.forEach(
+          (k, v) -> {
+            // Encode the ID into a websafe string
+            String encodedID = Base64.getUrlEncoder().encodeToString(String.valueOf(k).getBytes());
+
+            // Build up string with values from Cloud SQL
+            String recordOutput =
+                String.format(
+                    blogPostDisplayFormat,
+                    v.get("title"),
+                    v.get("publishTime"),
+                    v.get("author"),
+                    encodedID,
+                    encodedID,
+                    v.get("body"));
+
+            out.println(recordOutput); // print out the HTML
+          });
+
+    } catch (SQLException e) {
+      throw new ServletException("SQL error", e);
+    }
+  }
+
+  @Override
+  public void init() throws ServletException {
+    try {
+      String url = System.getProperty("cloudsql");
+
+      try {
+        conn = DriverManager.getConnection(url);
+
+        // Create the tables so that the SELECT query doesn't throw an exception
+        // if the user visits the page before any posts have been added
+
+        conn.createStatement().executeUpdate(createContentTableSql); // create content table
```

Comment:
You might wish to do this in it's own try, just in case there is already a table.

Reply:
Also, there is no guarantee this will run before any other init(), or doGet().